File: frontend/sockstore-frontend/src/app/models/order.ts
import { OrderItem } from "./order-item";

export interface Order {

    id?: number;
  
    customerName: string;
  
    address: string;
  
    phone: string;
  
    total: number;
  
    createdAt?: Date;

    status?: string;

    items: OrderItem[];
  
  }

File: backend/SockStoreAPI/Models/Order.cs
namespace SockStoreAPI.Models
{
    public class Order
    {
        public int Id { get; set; }

        public string CustomerName { get; set; } = string.Empty;

        public string Address { get; set; } = string.Empty;

        public string Phone { get; set; } = string.Empty;

        public decimal Total { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.Now;

        public string Status { get; set; } = "Pending";   

        public List<OrderItem> Items { get; set; } = new();

        public int UserId { get; set; }

        public User? User { get; set; }
    }
}